models/ExternalCalendarEntry.php

```diff
@@ -551,4 +552,23 @@ public function wasModifiedSince(ICalEventIF $icalEvent)
 
         return !$this->last_modified || CalendarUtils::formatDateTimeToAppTime($icalEvent->getLastModified()) > $this->getLastModifiedDateTime();
     }
+
+     /**
+     * Get location of this external calendar entry
+     *
+     * @return string
+     */
+    public function getLocation(bool $asHtml = false)
+    {
+        if (!$asHtml) {
+            return $this->location;
+        }
+        if (
+            filter_var($this->location, FILTER_VALIDATE_URL) !== false
+            && strpos($this->location, 'https://') === 0 // restrict to secure URLs (and not HTTP, SSF, FTP, etc.)
+        ) {
+            return Button::asLink($this->location)->link($this->location)->options(['target' => '_blank']);

```

Comment:
My only issue here would be that there doesn't seem to be a way of adding `noopener` or `noreferrer` attributes here? Maybe we should do this in the Button widget @luke-?

I believe on these lines we can add something like `['rel' => 'noopener noreferrer']` for added security and user privacy?
https://github.com/humhub/humhub/blob/master/protected/humhub/widgets/Button.php#L236-L238C20

Reply:
I would see this as a seperate issue as it should be implemented the same in the Calendar module and probably other places in HumHub.

Reply:
> I would see this as a seperate issue as it should be implemented the same in the Calendar module and probably other places in HumHub.

I agree.